### PR TITLE
update changelogs of 3.4 and 3.5 for handling auth invalid token and old revision errors in watch

### PR DESCRIPTION
**Pull request by mitake**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

Updating changelogs for https://github.com/etcd-io/etcd/pull/14547 and https://github.com/etcd-io/etcd/pull/14548 . cc @ahrtr @spzala @kafuu-chino

### CODE REVIEW

File: raft/rawnode.go [hunk 1]
```diff
@@ -172,9 +175,6 @@ func (rn *RawNode) HasReady() bool {
 // Advance notifies the RawNode that the application has applied and saved progress in the
 // last Ready results.
 func (rn *RawNode) Advance(rd Ready) {
-	if !IsEmptyHardState(rd.HardState) {
```
Comment (spzala) [inline raft/rawnode.go:175]: @mitake what are these changes related to? Thanks!

Comment (spzala) [reply raft/rawnode.go:175]: @mitake I see a second commit signed-by Nathan. I think we should make such changes separately on a clean PR? Thanks!

Comment (ahrtr) [reply raft/rawnode.go:175]: +1

@mitake you need to rebase this PR.

File: CHANGELOG/CHANGELOG-3.5.md [hunk 2]
```diff
@@ -4,6 +4,13 @@ Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/
 
 <hr>
 
+## v3.5.6 (TBD)
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.5.md:7]: @serathius Do you plan to release 3.5.6 or 3.6.0 in the next step? 3.5.6 seems better if there are enough bug fixes.

File: raft/rawnode.go (hunk 1 above)
Comment (mitake) [reply raft/rawnode.go]: @spzala @ahrtr oh sorry, I rebased on the latest main

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @mitake

File: raft/rawnode.go (hunk 1 above)
Comment (spzala) [reply raft/rawnode.go]: Thanks @mitake @ahrtr

**Comment by mitake** [comment]
Thanks @ahrtr @spzala !
